tests/config/test_config.py

```diff
@@ -116,3 +116,13 @@ def test_short_verbose(capsys: CaptureFixture) -> None:
     Run([str(EMPTY_MODULE), "-v"], exit=False)
     output = capsys.readouterr()
     assert "Using config file" in output.err
+
+
+def test_argument_separator(capsys: CaptureFixture) -> None:
+    """Check that we support using '--' to separate argument types.
+
+    Reported in https://github.com/PyCQA/pylint/issues/7003.
+    """
+    Run(["--", str(EMPTY_MODULE)], exit=False)
```

Comment:
Should we test that option are taken into account for example here an init-hook that print ?

Reply:
We could, but I'm fairly confident this should work. The code that this is changing is a `pylint` hack on top of `argparse`. We parse arguments with `parse_known_args` so `argparse` doesn't complain about unknown options and leaves that to us.
I simply didn't know that this is allowed so this wasn't excluded from the list of unknown options. `argparse` probably does dit automatically in the normale `parse_args`.